public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiHealth,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { get, isEmpty } from 'lodash';
+import { RouteComponentProps } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHideSideNavBar } from '../../../main/hooks/useHideSideNavBar';
+import { Listener } from '../../../../utils/utils';
+import { darkModeEnabled } from '../../../../utils/kibanaUtils';
+import { AppState } from '../../../../redux/reducers';
+import {
+  startDetector,
+  stopDetector,
+  deleteDetector,
+  getDetector,
+  getHistoricalDetectorList,
+} from '../../../../redux/reducers/ad';
+import { BREADCRUMBS } from '../../../../utils/constants';
+import { DETECTOR_STATE } from '../../../../../server/utils/constants';
+import { HistoricalDetectorConfig } from '../../components/HistoricalDetectorConfig/HistoricalDetectorConfig';
+import { HistoricalDetectorControls } from '../../components/HistoricalDetectorControls/HistoricalDetectorControls';
+import { EditHistoricalDetectorModal } from '../ActionModals/EditHistoricalDetectorModal/EditHistoricalDetectorModal';
+import { DeleteHistoricalDetectorModal } from '../ActionModals/DeleteHistoricalDetectorModal/DeleteHistoricalDetectorModal';
+import { AnomalyHistory } from '../../../DetectorResults/containers/AnomalyHistory';
+import { stateToColorMap } from '../../../utils/constants';
+import {
+  HISTORICAL_DETECTOR_RESULT_REFRESH_RATE,
+  HISTORICAL_DETECTOR_ACTION,
+  HISTORICAL_DETECTOR_STOP_THRESHOLD,
+} from '../../utils/constants';
+import {
+  getCallout,
+  waitForMs,
+  getRunningHistoricalDetectorCount,
+} from '../../utils/helpers';
+import { CoreStart } from '../../../../../../../src/core/public';
+import { CoreServicesContext } from '../../../../components/CoreServices/CoreServices';
+import { getClusterStats } from '../../../../redux/reducers/elasticsearch';
+import { GET_ALL_DETECTORS_QUERY_PARAMS } from '../../../utils/constants';
+
+export interface HistoricalDetectorRouterProps {
+  detectorId?: string;
+}
+interface HistoricalDetectorDetailProps
+  extends RouteComponentProps<HistoricalDetectorRouterProps> {}
+
+interface HistoricalDetectorDetailModalState {
+  isOpen: boolean;
+  action: HISTORICAL_DETECTOR_ACTION | undefined;
+}
+
+export const HistoricalDetectorDetail = (
+  props: HistoricalDetectorDetailProps
+) => {
+  const core = React.useContext(CoreServicesContext) as CoreStart;
+  const isDark = darkModeEnabled();
+  useHideSideNavBar(true, false);
+  const dispatch = useDispatch();
+  const detectorId: string = get(props, 'match.params.detectorId', '');
+
+  const adState = useSelector((state: AppState) => state.ad);
+  const allDetectors = adState.detectors;
+  const errorGettingDetectors = adState.errorMessage;
+  const detector = allDetectors[detectorId];
+  const monitors = useSelector((state: AppState) => state.alerting.monitors);
```

Comment:
We can't create monitor for historical detector. Why need to get monitors here?

Reply:
Good catch. Left over from copied existing code.